includes/tracks/events/class-wc-admin-setup-wizard-tracking.php

```diff
@@ -19,9 +19,54 @@ public static function init() {
 			return;
 		}
 
+		add_action( 'wc_setup_footer', array( __CLASS__, 'add_footer_scripts' ) );
+		add_action( 'admin_init', array( __CLASS__, 'track_marketing_signup' ), 1 );
 		self::add_step_save_events();
 	}
 
+	/**
+	 * Get the name of the current step.
+	 *
+	 * @return string
+	 */
+	public static function get_current_step() {
+		return isset( $_GET['step'] ) ? sanitize_key( $_GET['step'] ) : ''; // phpcs:ignore WordPress.Security.NonceVerification.NoNonceVerification
+	}
+
+	/**
+	 * Add footer scripts to OBW since it does not contain hooks for
+	 * wp_footer to allow the default methods of enqueuing scripts.
+	 */
+	public static function add_footer_scripts() {
+		echo '<script type="text/javascript" src="https://stats.wp.com/w.js?ver=' . gmdate( 'YW' ) . '"></script>'; // @codingStandardsIgnoreLine
+		wc_print_js();
+	}
+
+	/**
+	 * Track the marketing form on submit.
+	 */
+	public static function track_marketing_signup() {
+		if ( 'next_steps' !== self::get_current_step() ) {
+			return;
+		}
+
+		wc_enqueue_js(
+			"
+			var form = $( '.newsletter-form-email' ).closest( 'form' );
+			$( document ).on( 'submit', form, function() {
+				window.wcTracks.recordEvent( 'obw_marketing_signup' );
+			} );
+			$( '.wc-setup-content a' ).click( function trackNextScreen( e ) {
+				var properties = {
+					next_url: e.target.href,
+					button: e.target.textContent && e.target.textContent.trim()
+				};
+				window.wcTracks.recordEvent( 'obw_ready_next_step', properties );
```

Comment:
This might not fire in time for links not using `target="_blank"`.  I think we'll need to use a callback and wait to load the page until this event is complete.

Alternatively, we can add a referral check on pages outside of wc-setup to see if the referring page was wc-setup.  Though I'm not sure of any performance implications for checking this on every page.

```php
if ( ! wp_get_referer() ) {
    return;
}

$parsed_url = wp_parse_url( wp_get_referer() );
parse_str( $parsed_url['query'], $query ); 

if ( ! $query['page'] || 'wc-setup' !== $query['page'] ) {
    return;
}

// Track current URL and page title
```

Reply:
How about using `onclick` and returning true? `_tkq.push` gets executed and involves only synchronous code, so the event will be fired. 

```js
function trackNextScreen( e ) {
	var properties = {
		next_url: e.target.href,
		button: e.target.textContent && e.target.textContent.trim()
	};
	window.wcTracks.recordEvent( 'obw_ready_next_step', properties );
	return true;
}
$( '.wc-setup-content a' ).each( function( i, link ) {
	link.onclick = trackNextScreen;
} );
```